# Md. Rafsan Jamil
# Bowelar Analysis

In [52]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [53]:
df=pd.read_csv('wickets.csv')
df

Player       Span  Mat  Inns  Balls   Runs  Wkts    BBI  \
0   M Muralitharan (ICC/SL)  1992-2010  133   230  44039  18180   800   9/51   
1            SK Warne (AUS)  1992-2007  145   273  40705  17995   708   8/71   
2         JM Anderson (ENG)  2003-2021  166   309  35772  16827   632   7/42   
3          A Kumble (INDIA)  1990-2008  132   236  40850  18355   619  10/74   
4          GD McGrath (AUS)  1993-2007  124   243  29248  12186   563   8/24   
..                      ...        ...  ...   ...    ...    ...   ...    ...   
74        SCG MacGill (AUS)  1998-2008   44    85  11237   6038   208  8/108   
75    Saqlain Mushtaq (PAK)  1995-2004   49    86  14070   6206   208  8/164   
76         AME Roberts (WI)  1974-1983   47    90  11135   5174   202   7/54   
77            JA Snow (ENG)  1965-1976   49    93  12021   5387   202   7/40   
78         JR Thomson (AUS)  1972-1985   51    90  10535   5601   200   6/46   

       BBM    Ave  Econ    SR   5  10  
0   16/220  22.72  2.47  55.0  67  22  
1   12/128  25.41  2.65  57.4  37  10  
2    11/71  26.62  2.82  56.6  31   3  
3   14/149  29.65  2.69  65.9  35   8  
4    10/27  21.64  2.49  51.9  29   3  
..     ...    ...   ...   ...  ..  ..  
74  12/107  29.02  3.22  54.0  12   2  
75  10/155  29.83  2.64  67.6  13   3  
76  12/121  25.61  2.78  55.1  11   2  
77  10/142  26.66  2.68  59.5   8   1  
78   9/105  28.00  3.18  52.6   8   0  

[79 rows x 14 columns]

In [54]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 79 entries, 0 to 78
Data columns (total 14 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Player  79 non-null     object 
 1   Span    79 non-null     object 
 2   Mat     79 non-null     int64  
 3   Inns    79 non-null     int64  
 4   Balls   79 non-null     int64  
 5   Runs    79 non-null     int64  
 6   Wkts    79 non-null     int64  
 7   BBI     79 non-null     object 
 8   BBM     79 non-null     object 
 9   Ave     79 non-null     float64
 10  Econ    79 non-null     float64
 11  SR      79 non-null     float64
 12  5       79 non-null     int64  
 13  10      79 non-null     int64  
dtypes: float64(3), int64(7), object(4)
memory usage: 8.8+ KB


In [55]:
df.describe()

Mat        Inns         Balls          Runs        Wkts  \
count   79.000000   79.000000     79.000000     79.000000   79.000000   
mean    80.278481  145.126582  18664.367089   8609.670886  317.544304   
std     28.663806   51.324850   7210.105636   3090.383196  122.098300   
min     37.000000   67.000000   8785.000000   4846.000000  200.000000   
25%     60.500000  110.000000  13673.500000   6456.500000  230.000000   
50%     71.000000  129.000000  16498.000000   7742.000000  266.000000   
75%     93.000000  169.000000  21742.500000   9756.000000  374.500000   
max    166.000000  309.000000  44039.000000  18355.000000  800.000000   

             Ave       Econ         SR          5         10  
count  79.000000  79.000000  79.000000  79.000000  79.000000  
mean   27.474557   2.806203  59.217722  16.367089   2.797468  
std     3.656452   0.351686   9.352578   9.661153   3.235935  
min    20.940000   1.980000  41.200000   3.000000   0.000000  
25%    24.500000   2.600000  53.300000   9.500000   1.000000  
50%    28.000000   2.820000  57.400000  14.000000   2.000000  
75%    29.870000   3.065000  63.950000  20.500000   3.500000  
max    34.790000   3.460000  91.900000  67.000000  22.000000

In [6]:
df.shape

(79, 14)

In [7]:
df.columns

Index(['Player', 'Span', 'Mat', 'Inns', 'Balls', 'Runs', 'Wkts', 'BBI', 'BBM',
       'Ave', 'Econ', 'SR', '5', '10'],
      dtype='object')

# Removing BBI and BBM column

In [56]:
df.drop('BBI',axis=1,inplace=True)

In [57]:
df.drop('BBM',axis=1,inplace=True)

In [58]:
df.head()

Player       Span  Mat  Inns  Balls   Runs  Wkts    Ave  \
0  M Muralitharan (ICC/SL)  1992-2010  133   230  44039  18180   800  22.72   
1           SK Warne (AUS)  1992-2007  145   273  40705  17995   708  25.41   
2        JM Anderson (ENG)  2003-2021  166   309  35772  16827   632  26.62   
3         A Kumble (INDIA)  1990-2008  132   236  40850  18355   619  29.65   
4         GD McGrath (AUS)  1993-2007  124   243  29248  12186   563  21.64   

   Econ    SR   5  10  
0  2.47  55.0  67  22  
1  2.65  57.4  37  10  
2  2.82  56.6  31   3  
3  2.69  65.9  35   8  
4  2.49  51.9  29   3

In [13]:
df.shape

(79, 12)

In [14]:
df.describe()

Mat        Inns         Balls          Runs        Wkts  \
count   79.000000   79.000000     79.000000     79.000000   79.000000   
mean    80.278481  145.126582  18664.367089   8609.670886  317.544304   
std     28.663806   51.324850   7210.105636   3090.383196  122.098300   
min     37.000000   67.000000   8785.000000   4846.000000  200.000000   
25%     60.500000  110.000000  13673.500000   6456.500000  230.000000   
50%     71.000000  129.000000  16498.000000   7742.000000  266.000000   
75%     93.000000  169.000000  21742.500000   9756.000000  374.500000   
max    166.000000  309.000000  44039.000000  18355.000000  800.000000   

             Ave       Econ         SR          5         10  
count  79.000000  79.000000  79.000000  79.000000  79.000000  
mean   27.474557   2.806203  59.217722  16.367089   2.797468  
std     3.656452   0.351686   9.352578   9.661153   3.235935  
min    20.940000   1.980000  41.200000   3.000000   0.000000  
25%    24.500000   2.600000  53.300000   9.500000   1.000000  
50%    28.000000   2.820000  57.400000  14.000000   2.000000  
75%    29.870000   3.065000  63.950000  20.500000   3.500000  
max    34.790000   3.460000  91.900000  67.000000  22.000000

# Renaming Columns

In [59]:
# renaming the column names
df = df.rename(columns={'Mat':'Match', 
                        'Inns':'Innings',
                        'Wkts': 'Wickets',
                        'SR': 'Strike Rate',
                        'Ave': 'Average',
                        'Econ': 'Economy'
                        })

# splitting the 'Player' column to get the information about 'Country'
df_player = df['Player'].str.split("(", expand=True)

# concatinating 'Country' with the main dataframe
df = pd.concat([df, df_player], axis=1)

# dropping the 'Player' columns
df = df.drop('Player', axis=1)

# renaming the column names
df = df.rename(columns={0: 'Player',
                        1: 'Country'})

# remove the ")" from the 'Country' column
df['Country'] = df['Country'].str.replace(")", "")

# rearrange the columns
#new_col_sequence = ['Player', 'Span', 'Match', 'Innings', 'Balls', 'Runs', 'Wickets',
                    #'Average', 'Economy', 'Strike Rate',5,10, 'Country']
#df = df[new_col_sequence]

display(df.head())

Span  Match  Innings  Balls   Runs  Wickets  Average  Economy  \
0  1992-2010    133      230  44039  18180      800    22.72     2.47   
1  1992-2007    145      273  40705  17995      708    25.41     2.65   
2  2003-2021    166      309  35772  16827      632    26.62     2.82   
3  1990-2008    132      236  40850  18355      619    29.65     2.69   
4  1993-2007    124      243  29248  12186      563    21.64     2.49   

   Strike Rate   5  10           Player Country  
0         55.0  67  22  M Muralitharan   ICC/SL  
1         57.4  37  10        SK Warne      AUS  
2         56.6  31   3     JM Anderson      ENG  
3         65.9  35   8        A Kumble    INDIA  
4         51.9  29   3      GD McGrath      AUS

# Numbers of players played for ICC

In [60]:
def icc_check(x):
    if "ICC" in x:
        return "Yes"
    else:
        return "No"

In [61]:
df['played_for_ICC'] = df['Country'].apply(icc_check)
display(df.head(10))
df['played_for_ICC'].value_counts()

Span  Match  Innings  Balls   Runs  Wickets  Average  Economy  \
0  1992-2010    133      230  44039  18180      800    22.72     2.47   
1  1992-2007    145      273  40705  17995      708    25.41     2.65   
2  2003-2021    166      309  35772  16827      632    26.62     2.82   
3  1990-2008    132      236  40850  18355      619    29.65     2.69   
4  1993-2007    124      243  29248  12186      563    21.64     2.49   
5  2007-2021    149      274  29863  14590      524    27.84     2.93   
6  1984-2001    132      242  30019  12688      519    24.44     2.53   
7  2004-2019     93      171  18608  10077      439    22.95     3.24   
8  1978-1994    131      227  27740  12867      434    29.64     2.78   
9  1999-2018     93      170  25993  12157      433    28.07     2.80   

   Strike Rate   5  10           Player Country played_for_ICC  
0         55.0  67  22  M Muralitharan   ICC/SL            Yes  
1         57.4  37  10        SK Warne      AUS             No  
2         56.6  31   3     JM Anderson      ENG             No  
3         65.9  35   8        A Kumble    INDIA             No  
4         51.9  29   3      GD McGrath      AUS             No  
5         56.9  18   3       SCJ Broad      ENG             No  
6         57.8  22   3        CA Walsh       WI             No  
7         42.3  26   5        DW Steyn       SA             No  
8         63.9  23   2     N Kapil Dev    INDIA             No  
9         60.0  34   9    HMRKB Herath       SL             No

No     74
Yes     5
Name: played_for_ICC, dtype: int64

# Observation 
5 Players played for ICC

In [62]:
df['Country'] = df['Country'].str.replace("ICC/", "")

display(df.head())

Span  Match  Innings  Balls   Runs  Wickets  Average  Economy  \
0  1992-2010    133      230  44039  18180      800    22.72     2.47   
1  1992-2007    145      273  40705  17995      708    25.41     2.65   
2  2003-2021    166      309  35772  16827      632    26.62     2.82   
3  1990-2008    132      236  40850  18355      619    29.65     2.69   
4  1993-2007    124      243  29248  12186      563    21.64     2.49   

   Strike Rate   5  10           Player Country played_for_ICC  
0         55.0  67  22  M Muralitharan       SL            Yes  
1         57.4  37  10        SK Warne      AUS             No  
2         56.6  31   3     JM Anderson      ENG             No  
3         65.9  35   8        A Kumble    INDIA             No  
4         51.9  29   3      GD McGrath      AUS             No

In [63]:
df['Country'].value_counts()

AUS        18
ENG        13
INDIA      10
WI          9
SA          8
NZ          7
PAK         7
SL          3
ENG/ICC     2
BDESH       1
ZIM         1
Name: Country, dtype: int64

# Observation
18 Players played for Australia

1 Player played for Bangladesh

11 different countries players are in this dataset

In [64]:
 def Bangladesh_check(x):
    if "Bangladesh" in x:
        return 1
    else:
        return 0

In [65]:
df['played_for_Bangladesh'] = df['Country'].apply(Bdesh_check)

In [66]:
display(df.head(10))
df['played_for_Bangladesh'].value_counts()

Span  Match  Innings  Balls   Runs  Wickets  Average  Economy  \
0  1992-2010    133      230  44039  18180      800    22.72     2.47   
1  1992-2007    145      273  40705  17995      708    25.41     2.65   
2  2003-2021    166      309  35772  16827      632    26.62     2.82   
3  1990-2008    132      236  40850  18355      619    29.65     2.69   
4  1993-2007    124      243  29248  12186      563    21.64     2.49   
5  2007-2021    149      274  29863  14590      524    27.84     2.93   
6  1984-2001    132      242  30019  12688      519    24.44     2.53   
7  2004-2019     93      171  18608  10077      439    22.95     3.24   
8  1978-1994    131      227  27740  12867      434    29.64     2.78   
9  1999-2018     93      170  25993  12157      433    28.07     2.80   

   Strike Rate   5  10           Player Country played_for_ICC  \
0         55.0  67  22  M Muralitharan       SL            Yes   
1         57.4  37  10        SK Warne      AUS             No   
2         56.6  31   3     JM Anderson      ENG             No   
3         65.9  35   8        A Kumble    INDIA             No   
4         51.9  29   3      GD McGrath      AUS             No   
5         56.9  18   3       SCJ Broad      ENG             No   
6         57.8  22   3        CA Walsh       WI             No   
7         42.3  26   5        DW Steyn       SA             No   
8         63.9  23   2     N Kapil Dev    INDIA             No   
9         60.0  34   9    HMRKB Herath       SL             No   

   played_for_Bangladesh  
0                      0  
1                      0  
2                      0  
3                      0  
4                      0  
5                      0  
6                      0  
7                      0  
8                      0  
9                      0

0    79
Name: played_for_Bangladesh, dtype: int64

In [67]:
# splitting the 'Span' column based on the "-"
df_span = df['Span'].str.split("-", expand=True)

# conccatinating the new dataframe with the main dataframe
df = pd.concat([df, df_span], axis=1)

# renaming the newly created column names
df = df.rename(columns={0: "start_year",
                        1: "end_year"})

# removing the "Span" column
df = df.drop("Span", axis=1)

display(df.head())

Match  Innings  Balls   Runs  Wickets  Average  Economy  Strike Rate   5  \
0    133      230  44039  18180      800    22.72     2.47         55.0  67   
1    145      273  40705  17995      708    25.41     2.65         57.4  37   
2    166      309  35772  16827      632    26.62     2.82         56.6  31   
3    132      236  40850  18355      619    29.65     2.69         65.9  35   
4    124      243  29248  12186      563    21.64     2.49         51.9  29   

   10           Player Country played_for_ICC  played_for_Bangladesh  \
0  22  M Muralitharan       SL            Yes                      0   
1  10        SK Warne      AUS             No                      0   
2   3     JM Anderson      ENG             No                      0   
3   8        A Kumble    INDIA             No                      0   
4   3      GD McGrath      AUS             No                      0   

  start_year end_year  
0       1992     2010  
1       1992     2007  
2       2003     2021  
3       1990     2008  
4       1993     2007

In [68]:

df['start_year'] = df['start_year'].astype('int') 
df['end_year'] = df['end_year'].astype('int')

print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 79 entries, 0 to 78
Data columns (total 16 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Match                  79 non-null     int64  
 1   Innings                79 non-null     int64  
 2   Balls                  79 non-null     int64  
 3   Runs                   79 non-null     int64  
 4   Wickets                79 non-null     int64  
 5   Average                79 non-null     float64
 6   Economy                79 non-null     float64
 7   Strike Rate            79 non-null     float64
 8   5                      79 non-null     int64  
 9   10                     79 non-null     int64  
 10  Player                 79 non-null     object 
 11  Country                79 non-null     object 
 12  played_for_ICC         79 non-null     object 
 13  played_for_Bangladesh  79 non-null     int64  
 14  start_year             79 non-null     int32  
 15  end_year

In [69]:
df['years_played'] = df['end_year'] - df['start_year']

df = df.drop(['start_year', "end_year"], axis=1)

display(df.head(10))


Match  Innings  Balls   Runs  Wickets  Average  Economy  Strike Rate   5  \
0    133      230  44039  18180      800    22.72     2.47         55.0  67   
1    145      273  40705  17995      708    25.41     2.65         57.4  37   
2    166      309  35772  16827      632    26.62     2.82         56.6  31   
3    132      236  40850  18355      619    29.65     2.69         65.9  35   
4    124      243  29248  12186      563    21.64     2.49         51.9  29   
5    149      274  29863  14590      524    27.84     2.93         56.9  18   
6    132      242  30019  12688      519    24.44     2.53         57.8  22   
7     93      171  18608  10077      439    22.95     3.24         42.3  26   
8    131      227  27740  12867      434    29.64     2.78         63.9  23   
9     93      170  25993  12157      433    28.07     2.80         60.0  34   

   10           Player Country played_for_ICC  played_for_Bangladesh  \
0  22  M Muralitharan       SL            Yes                      0   
1  10        SK Warne      AUS             No                      0   
2   3     JM Anderson      ENG             No                      0   
3   8        A Kumble    INDIA             No                      0   
4   3      GD McGrath      AUS             No                      0   
5   3       SCJ Broad      ENG             No                      0   
6   3        CA Walsh       WI             No                      0   
7   5        DW Steyn       SA             No                      0   
8   2     N Kapil Dev    INDIA             No                      0   
9   9    HMRKB Herath       SL             No                      0   

   years_played  
0            18  
1            15  
2            18  
3            18  
4            14  
5            14  
6            17  
7            15  
8            16  
9            19

In [70]:
df.sort_values(by='years_played', ascending = False).head(10)

Match  Innings  Balls   Runs  Wickets  Average  Economy  Strike Rate   5  \
21     88      142  19458   8258      362    22.81     2.54         53.7  23   
55     93      159  21599   7999      235    34.03     2.22         91.9   6   
9      93      170  25993  12157      433    28.07     2.80         60.0  34   
0     133      230  44039  18180      800    22.72     2.47         55.0  67   
2     166      309  35772  16827      632    26.62     2.82         56.6  31   
3     132      236  40850  18355      619    29.65     2.69         65.9  35   
37    166      272  20232   9535      292    32.65     2.82         69.2   5   
33     79      148  27115   8989      309    29.09     1.98         87.7  18   
13    104      181  22627   9779      414    23.62     2.59         54.6  25   
22    113      187  28814  12441      362    34.36     2.59         79.5  20   

    10           Player Country played_for_ICC  played_for_Bangladesh  \
21   6      Imran Khan      PAK             No                      0   
55   0       GS Sobers       WI             No                      0   
9    9    HMRKB Herath       SL             No                      0   
0   22  M Muralitharan       SL            Yes                      0   
2    3     JM Anderson      ENG             No                      0   
3    8        A Kumble    INDIA             No                      0   
37   0       JH Kallis       SA            Yes                      0   
33   2        LR Gibbs       WI             No                      0   
13   5     Wasim Akram      PAK             No                      0   
22   3      DL Vettori       NZ            Yes                      0   

    years_played  
21            21  
55            20  
9             19  
0             18  
2             18  
3             18  
37            18  
33            18  
13            17  
22            17

In [71]:
df.sort_values(by='years_played', ascending = True).head(10)

Match  Innings  Balls  Runs  Wickets  Average  Economy  Strike Rate   5  \
44     60      109  15349  7642      255    29.96     2.98         60.1  17   
71     47       86   8785  4846      213    22.75     3.30         41.2  10   
72     55      103  11887  5438      212    25.65     2.74         56.0   9   
63     63      115  13375  7192      226    31.82     3.22         59.1   8   
54     46       85  13697  7306      235    31.08     3.20         58.2  16   
28     73      140  16001  8891      313    28.40     3.33         51.1  12   
48     67      122  13909  7564      248    30.50     3.26         56.0   7   
59     54      102  11991  6046      229    26.40     3.02         52.3   9   
60     58       95  11821  6503      229    28.39     3.30         51.6   9   
51     51       92  15918  5876      236    24.89     2.21         67.4  15   

    10         Player  Country played_for_ICC  played_for_Bangladesh  \
44   3      GP Swann       ENG             No                      0   
71   4      K Rabada        SA             No                      0   
72   0  JR Hazlewood       AUS             No                      0   
63   1   SJ Harmison   ENG/ICC            Yes                      0   
54   3    Yasir Shah       PAK             No                      0   
28   3    MG Johnson       AUS             No                      0   
48   1    MJ Hoggard       ENG             No                      0   
59   0      N Wagner        NZ             No                      0   
60   0       D Gough       ENG             No                      0   
51   5     AV Bedser       ENG             No                      0   

    years_played  
44             5  
71             6  
72             7  
63             7  
54             7  
28             8  
48             8  
59             9  
60             9  
51             9

# Observation
1.Imran Khan from Pakistan played the longest period

2.GP Swann from England played the shortest period

In [72]:
df.sort_values(by='Economy', ascending = True).head(10)

Match  Innings  Balls  Runs  Wickets  Average  Economy  Strike Rate   5  \
33     79      148  27115  8989      309    29.09     1.98         87.7  18   
47     63      116  19108  6704      248    27.03     2.10         77.0  16   
35     86      151  21862  7674      297    25.83     2.10         73.6  17   
39     67      118  21364  7637      266    28.71     2.14         80.3  14   
68     37       67  14513  5231      216    24.21     2.16         67.1  21   
51     51       92  15918  5876      236    24.89     2.21         67.4  15   
55     93      159  21599  7999      235    34.03     2.22         91.9   6   
61     61      113  13650  5251      228    23.03     2.30         59.8  12   
15     98      179  22103  8501      405    20.99     2.30         54.5  22   
45     70      129  16056  6261      252    24.84     2.33         63.7   9   

    10         Player Country played_for_ICC  played_for_Bangladesh  \
33   2      LR Gibbs       WI             No                      0   
47   1      R Benaud      AUS             No                      0   
35   6  DL Underwood      ENG             No                      0   
39   1       BS Bedi    INDIA             No                      0   
68   7   CV Grimmett      AUS             No                      0   
51   5     AV Bedser      ENG             No                      0   
55   0     GS Sobers       WI             No                      0   
61   0   RR Lindwall      AUS             No                      0   
15   3   CEL Ambrose       WI             No                      0   
45   1    JB Statham      ENG             No                      0   

    years_played  
33            18  
47            12  
35            16  
39            13  
68            11  
51             9  
55            20  
61            14  
15            12  
45            14

In [73]:
df.sort_values(by='Strike Rate', ascending = True).head(10)

Match  Innings  Balls   Runs  Wickets  Average  Economy  Strike Rate   5  \
71     47       86   8785   4846      213    22.75     3.30         41.2  10   
7      93      171  18608  10077      439    22.95     3.24         42.3  26   
20     87      154  16224   8788      373    23.56     3.25         43.4  22   
19     81      151  17584   7876      376    20.94     2.68         46.7  22   
25     72      129  15519   7344      330    22.25     2.83         47.0  20   
43     61      117  12575   7031      255    27.57     3.35         49.3  13   
34     67      127  15178   6625      307    21.57     2.61         49.4  17   
65     64      119  11391   5000      224    22.32     2.63         50.8  13   
10     86      150  21918   9611      431    22.29     2.63         50.8  36   
41     58      111  13169   5433      259    20.97     2.47         50.8   7   

    10          Player Country played_for_ICC  played_for_Bangladesh  \
71   4       K Rabada       SA             No                      0   
7    5       DW Steyn       SA             No                      0   
20   5   Waqar Younis      PAK             No                      0   
19   4    MD Marshall       WI             No                      0   
25   3      AA Donald       SA             No                      0   
43   2       MA Starc      AUS             No                      0   
34   3     FS Trueman      ENG             No                      0   
65   2   VD Philander       SA             No                      0   
10   9  Sir RJ Hadlee       NZ             No                      0   
41   0       J Garner       WI             No                      0   

    years_played  
71             6  
7             15  
20            14  
19            13  
25            10  
43            10  
34            13  
65             9  
10            17  
41            10

In [75]:
df.sort_values(by='Average', ascending = True).head(10)

Match  Innings  Balls   Runs  Wickets  Average  Economy  Strike Rate   5  \
19     81      151  17584   7876      376    20.94     2.68         46.7  22   
41     58      111  13169   5433      259    20.97     2.47         50.8   7   
15     98      179  22103   8501      405    20.99     2.30         54.5  22   
34     67      127  15178   6625      307    21.57     2.61         49.4  17   
4     124      243  29248  12186      563    21.64     2.49         51.9  29   
25     72      129  15519   7344      330    22.25     2.83         47.0  20   
10     86      150  21918   9611      431    22.29     2.63         50.8  36   
65     64      119  11391   5000      224    22.32     2.63         50.8  13   
0     133      230  44039  18180      800    22.72     2.47         55.0  67   
71     47       86   8785   4846      213    22.75     3.30         41.2  10   

    10           Player Country played_for_ICC  played_for_Bangladesh  \
19   4     MD Marshall       WI             No                      0   
41   0        J Garner       WI             No                      0   
15   3     CEL Ambrose       WI             No                      0   
34   3      FS Trueman      ENG             No                      0   
4    3      GD McGrath      AUS             No                      0   
25   3       AA Donald       SA             No                      0   
10   9   Sir RJ Hadlee       NZ             No                      0   
65   2    VD Philander       SA             No                      0   
0   22  M Muralitharan       SL            Yes                      0   
71   4        K Rabada       SA             No                      0   

    years_played  
19            13  
41            10  
15            12  
34            13  
4             14  
25            10  
10            17  
65             9  
0             18  
71             6

# Observation 
1.LR Gibbs from West Indies has the lowest Economy Rate of 1.98

2.K Rabada from South Africa has the lowest Strike  Rate of 41.2

3.MD Marshall from West Indies has the lowest Average of 20.94